storage/innobase/xtrabackup/src/keyring_plugins.cc

```diff
@@ -74,17 +74,28 @@ static void init_plugins(int argc, char **argv) {
   memcpy(t_argv, argv, sizeof(char *) * t_argc);
 
   mysql_optional_plugins[0] = 0;
-
-    /* initialize only daemon_keyring_proxy_plugin */
-    set_srv_keyring_implementation_as_default();
-    for (st_mysql_plugin **plugin = mysql_mandatory_plugins; *plugin;
-         plugin++) {
-      if (strcmp((*plugin)->name, "daemon_keyring_proxy_plugin") == 0) {
-        mysql_mandatory_plugins[0] = *plugin;
-        mysql_mandatory_plugins[1] = 0;
-        break;
+  I_List_iterator<i_string> iter(*opt_plugin_load_list_ptr);
+  i_string *item;
+  while (nullptr != (item = iter++)) {
+    const size_t prefix_len = sizeof("keyring_") - 1;
+    const char *plugin_name = item->ptr;
+    if (strlen(plugin_name) > prefix_len &&
+        strncmp(plugin_name, "keyring_", prefix_len) == 0) {
+      /* initialize only daemon_keyring_proxy_plugin */
+      set_srv_keyring_implementation_as_default();
+      for (st_mysql_plugin **plugin = mysql_mandatory_plugins; *plugin;
+           plugin++) {
+        if (strcmp((*plugin)->name, "daemon_keyring_proxy_plugin") == 0) {
+          mysql_mandatory_plugins[0] = *plugin;
+          mysql_mandatory_plugins[1] = 0;
```

Comment:
will this be initialized multiple times? if both keyring_file/keyring_component is passed?
Maybe passing both is invalid? :thinking: